From 4bc43b66a357353c414ba7bdb0c8210b53c4eb68 Mon Sep 17 00:00:00 2001
From: Aditya Garg <gargaditya08@live.com>
Date: Tue, 31 Dec 2024 20:08:49 +0530
Subject: [PATCH] backlight

---
 drivers/hid/hid-appletb-kbd.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/hid/hid-appletb-kbd.c b/drivers/hid/hid-appletb-kbd.c
index ecc8c84f4..b07c394fb 100644
--- a/drivers/hid/hid-appletb-kbd.c
+++ b/drivers/hid/hid-appletb-kbd.c
@@ -497,6 +497,9 @@ static struct hid_driver appletb_kbd_hid_driver = {
 };
 module_hid_driver(appletb_kbd_hid_driver);
 
+/* The backlight driver should be loaded before the keyboard driver is initialised*/
+MODULE_SOFTDEP("pre: hid_appletb_bl");
+
 MODULE_AUTHOR("Ronald Tschalär");
 MODULE_AUTHOR("Kerem Karabay <kekrby@gmail.com>");
 MODULE_DESCRIPTION("MacBookPro Touch Bar Keyboard Mode Driver");
-- 
2.43.0

