From e143fd17ffa6736a61159b4c32b287cbe3b97a7e Mon Sep 17 00:00:00 2001
From: Aditya Garg <85610623+AdityaGarg8@users.noreply.github.com>
Date: Tue, 14 Nov 2023 14:35:48 +0530
Subject: [PATCH] Update hci_bcm4377.c

---
 drivers/bluetooth/hci_bcm4377.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/bluetooth/hci_bcm4377.c b/drivers/bluetooth/hci_bcm4377.c
index a617578356953c..bc8398cde3c30d 100644
--- a/drivers/bluetooth/hci_bcm4377.c
+++ b/drivers/bluetooth/hci_bcm4377.c
@@ -2453,6 +2453,7 @@ static const struct bcm4377_hw bcm4377_hw_variants[] = {
 		.bar0_window2 = 0x1810c000,
 		.disable_aspm = true,
 		.broken_ext_scan = true,
+		.broken_le_coded = true,
 		.send_ptb = bcm4377_send_ptb,
 	},
 
